spec/solver_spec.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Test Solver class and its methods' do
+    before(:each) do
+      @solve = Solver.new
+    end
```

Comment:
- [ ] You have done a great job adding most of the test cases, but you missed the #factorial method test that handles a negative value.  It should throw an exception.  Kindly add that test case.

Reply:
Thanks for review